poshc2/client/command_handlers/ImplantHandler.py

```diff
@@ -477,6 +491,49 @@ def do_show_urls(user, command):
     clear()
 
 
+def do_show_hosted_files(user, command):
+    files = update_cache_urls()
+    filesformatted = "ID  URI  FilePath  ContentType  Base64  Active\n"
+    for i in files:
+        filesformatted += "%s  %s  %s  %s  %s %s \n" % (i[0], i[1], i[2], i[3], i[4], i[5])
+    print_good(filesformatted)
+    input("Press Enter to continue...")
+    clear()
+
+
+def do_add_hosted_file(user, command):
+    URI = input("URI Path: downloads/2020/application.docx ")
+    FilePath = input("File Path: .e.g. /tmp/application.docx ")
+    ContentType = input("Content Type: .e.g. text/html ")
+    Base64 = input("Base64 Encode File (Yes/No): ")
+    insert_hosted_file(URI, FilePath, ContentType, Base64, "Yes")
```

Comment:
Genuine question, is it better storing the filepath and loading the file each time or loading and storing the file in the db and serving the same one each time? :thinking:

Reply:
Maybe, could be an update down the line. I suppose its really not a bad shout tho